src/main/java/com/redhat/cloud/notifications/db/ApplicationResources.java

```diff
@@ -48,9 +48,13 @@
     }
 
     public Uni<Boolean> deleteApplication(UUID applicationId) {
-        String query = "DELETE FROM public.applications WHERE id = $1";
+        // We need to first delete endpoints that point to
+        // event types that are linked to this application
 
-        return runDeleteQuery(applicationId, query);
+        String ep_query = "DELETE FROM endpoint_targets et USING event_type ev  WHERE et.event_type_id = ev.id AND ev.application_id = $1";
```

Comment:
You won't need to do this anymore after: https://github.com/RedHatInsights/notifications-backend/pull/110/files#diff-cfb416495b160128789c37c83b6a95cd34abc5818859735bcb5354492b9fe51cR20-R24